components/org.wso2.carbon.privacy.forgetme.analytics.streams/src/main/java/org/wso2/carbon/privacy/forgetme/analytics/streams/instructions/AnalyticsStreamsInstruction.java

```diff
@@ -0,0 +1,176 @@
+    private static Object hashRecordValue(Object value) {
+        return Objects.hash(value);
+    }
+
+    @Override
+    public ForgetMeResult execute(UserIdentifier userIdentifier, ProcessorConfig processorConfig,
+                                  Environment environment, ReportAppender reportAppender)
+            throws InstructionExecutionException {
+        AnalyticsDataService analyticsDataService = getAnalyticsDataService();
+        for (Streams.Stream stream : this.streams) {
+            filterRecords(analyticsDataService, userIdentifier, stream);
+        }
+
+        try {
+            updateQueuedRecords(analyticsDataService);
+        } catch (AnalyticsException e) {
+            throw new AnalyticsStreamsProcessorException("Error in updating data records.", e);
+        }
+        return new ForgetMeResult();
+    }
+
+    /**
+     * Get analytics data access layer.
+     *
+     * @return
+     */
+    private AnalyticsDataService getAnalyticsDataService() {
+        // This requires "wso2_custom_conf_dir" property to point to the target DAS pack's conf directory.
+        String carbonHome = environment.getProperty(CARBON_HOME);
+        System.setProperty(CUSTOM_CONF_DIR_NAME, Paths.get(carbonHome, CUSTOM_CONF_DIR_PATH_SEGMENT).toString());
+
+        AnalyticsServiceHolder.setAnalyticsClusterManager(new AnalyticsClusterManagerImpl());
+        return AnalyticsServiceHolder.getAnalyticsDataService();
+    }
+
+    /**
+     * Filter records which are to be anonymized.
+     *
+     * @param analyticsDataService Analytics data service reference
+     * @param userIdentifier       User identifier
+     * @param stream               Stream
+     */
+    private void filterRecords(AnalyticsDataService analyticsDataService, UserIdentifier userIdentifier,
+                               Streams.Stream stream) {
+        try {
+            AnalyticsDataResponse analyticsDataResponse = analyticsDataService.get(userIdentifier.getTenantId(),
+                    getTableName(stream.getStreamName()), 1, null, Long.MIN_VALUE, Long.MAX_VALUE, 0,
+                    RECORD_BATCH_SIZE);
+            List<AnalyticsDataResponse.Entry> entries = analyticsDataResponse.getEntries();
+
+            for (AnalyticsDataResponse.Entry entry : entries) {
+                AnalyticsIterator<Record> recordAnalyticsIterator = analyticsDataService.readRecords(
+                        entry.getRecordStoreName(), entry.getRecordGroup());
+
+                while (recordAnalyticsIterator.hasNext()) {
+                    Record record = recordAnalyticsIterator.next();
+                    if (record.getValue(stream.getId()).equals(userIdentifier.getUsername())) {
+                        this.anonymizedRecords.add(anonymizeRecord(record, stream, userIdentifier.getPseudonym()));
+                    }
+                }
+            }
+        } catch (AnalyticsException e) {
+            logger.error("Error occurred while filtering record set.", e);
```

Comment:
Don't we need to break the flow here on an exception ?

Reply:
No. Since we are reading stream-by-stream here, inability to read a stream should not break the entire flow.